apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/TraceContext.java

```diff
@@ -199,6 +199,8 @@ public boolean asChildOf(String traceParentHeader) {
         } catch (IllegalArgumentException e) {
             logger.warn(e.getMessage());
             return false;
+        } finally {
+            onMutation();
```

Comment:
I think it is a bug without that, right?

Reply:
Is that when re-using a TraceContext? Could you add a test case for this scenario?

Reply:
No, it's when I set the Transaction a child of remote span after it is created. The tests pass without it, but I noticed in the logs that it was wrong. Will add a test.